### PR TITLE
Fix/wp review updates

### PR DESCRIPTION
**Pull request by ssolders**
Wordpress review of the plugin left a few remarks.

## 1) Incorrect Stable Tag
I copied what Klarna did, that the latest stable version was "trunk" which apparently isn't allowed.
Changed to the latest version we also specify in 3-4 other places (latest with this update is 1.1.2

## 2) Data Must be Sanitized, Escaped, and Validated
Data received from requests must be sanitzed properly

## 3) Variables and options must be escaped when echo'd
Variables we print out in html-markup must be escaped

### CODE REVIEW

File: classes/class-ledyer-ajax.php [hunk 1]
```diff
@@ -60,8 +60,11 @@ public static function lco_wc_update_cart() {
 
 		$values = array();
 		if ( isset( $_POST['checkout'] ) ) {
-			parse_str( wp_unslash( $_POST['checkout'] ), $values );// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			$response = wp_unslash( $_POST['checkout'] );
+			parse_str( html_entity_decode($response), $values );
+			$values=filter_var_array($values, FILTER_SANITIZE_ENCODED);
```
Comment (ssolders) [inline classes/class-ledyer-ajax.php:65]: Previously we instantly converted the received string array of url query params to an array.

Updated to first make the request, stored the response, which is the url query parameter string as `$response`.
Then use the built in `html_entity_decode` function to remove any illegal inputs. After that, parse the string to an array.

Finally filter the array to sanitize is.

File: classes/class-ledyer-lco-gateway.php [hunk 2]
```diff
@@ -400,11 +400,11 @@ public function ledyer_order_billing_fileds( $order ) {
 			$care_of = get_post_meta($order_id, '_billing_care_of', true);
 
 			if( ! empty( $attention_name ) ) {
-				echo '<p><strong>' . __('Billing Attention Name') . ':</strong><br> ' . $attention_name . '</p>';
+				echo '<p><strong>' . __('Billing Attention Name') . ':</strong><br> ' . esc_html($attention_name) . '</p>';
```
Comment (ssolders) [inline classes/class-ledyer-lco-gateway.php:403]: Just escape the variables printed out in html

File: classes/class-ledyer-lco-gateway.php [hunk 3]
```diff
@@ -400,11 +400,11 @@ public function ledyer_order_billing_fileds( $order ) {
...
 			}
 
 			if( ! empty( $care_of ) ) {
-				echo '<p><strong>' . __('Billing Care Of') . ':</strong><br> ' . $care_of. '</p>';
+				echo '<p><strong>' . __('Billing Care Of') . ':</strong><br> ' . esc_html($care_of) . '</p>';
```
Comment (ssolders) [inline classes/class-ledyer-lco-gateway.php:407]: Just escape the variables printed out in html

File: classes/class-ledyer-lco-gateway.php [hunk 4]
```diff
@@ -425,11 +425,11 @@ public function ledyer_order_shipping_fileds( $order ) {
 			$care_of        = get_post_meta( $order_id, '_shipping_care_of', true );
 
 			if ( ! empty( $attention_name ) ) {
-				echo '<p><strong>' . __( 'Shipping Attention Name' ) . ':</strong><br> ' . $attention_name . '</p>';
+				echo '<p><strong>' . __( 'Shipping Attention Name' ) . ':</strong><br> ' . esc_html($attention_name) . '</p>';
```
Comment (ssolders) [inline classes/class-ledyer-lco-gateway.php:428]: Just escape the variables printed out in html

File: classes/class-ledyer-lco-gateway.php [hunk 5]
```diff
@@ -425,11 +425,11 @@ public function ledyer_order_shipping_fileds( $order ) {
...
 			}
 
 			if ( ! empty( $care_of ) ) {
-				echo '<p><strong>' . __( 'Shipping Care Of' ) . ':</strong><br> ' . $care_of . '</p>';
+				echo '<p><strong>' . __( 'Shipping Care Of' ) . ':</strong><br> ' . esc_html($care_of) . '</p>';
```
Comment (ssolders) [inline classes/class-ledyer-lco-gateway.php:432]: Just escape the variables printed out in html

File: classes/class-ledyer-ajax.php (hunk 1 above)
Comment (perjo927) [reply classes/class-ledyer-ajax.php:65]: Gotta love the fact that it was consciously ignored before ❤️ 
`// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized`

**Review by perjo927 — APPROVED** [review]
LGTM!

**Review by timjackleus — APPROVED** [review]

**Comment by perjo927** [comment]
Couldn't it be nice to keep the test files for later?

**Pull request merged**